packages/core/types/index.d.ts

```diff
@@ -36,20 +34,15 @@ export function withEmotionCache<Props, RefType = any>(
 
 export const jsx: typeof createElement
 
-export type InterpolationWithTheme<Theme> =
-  | Interpolation
-  | ((theme: Theme) => Interpolation)
-
-export interface GlobalProps<Theme> {
-  styles: InterpolationWithTheme<Theme>
+export interface GlobalProps {
+  styles: Interpolation
```

Comment:
I believe this should be `Interpolation<Emotion.Theme>`. With the proposed approach here, in this PR, you indeed don't need to parametrize `GlobalProps` with `Theme` but you still need to provide the default theme type~ to `Interpolation` here (and probably in some other types where you have removed this).

Reply:
`Interpolation` defaults to `{ theme: Emotion.Theme }` so should be fine.

Reply:
Actually it doesn't from what I can see - https://github.com/emotion-js/emotion/blob/b81d652e70c8c421aae24e8fb357be6ff16143f0/packages/serialize/types/index.d.ts#L68

In here (for a `Global` case) it should be just `Interpolation<Emotion.Theme>` and not `Interpolation<{ theme: Emotion.Theme }>`

Reply:
Check out https://github.com/joltmode/emotion/pull/1/files#diff-e68603b09cd6bb7bbc9c8690f15e5392

I updated those types as part of this. But I didn't spend too much time on them, more putting this up as a general idea for joltmode if he wanted to go this way :)

Reply:
Gotcha - I think I was looking as original PR @joltmode looking for Interpolation changes 👍 

The point about `Global` stands though.

Reply:
Gotcha - I think I was looking as original PR @joltmode looking for Interpolation changes 👍 

The point about `Global` stands though.

Reply:
Dang, I get it. Good catch

Reply:
Fixed